dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/datavalueset/DataValueSetImportValidator.java

```diff
@@ -366,14 +366,13 @@ private static void validateDataValueIsDefined( DataValue dataValue, ImportConte
         }
     }
 
-    private static void validateDataValueIsValid( DataValue dataValue, ImportContext context,
+    private static void validateDataValueIsValid( DataValueEntry dataValue, ImportContext context,
         DataSetContext dataSetContext, DataValueContext valueContext )
     {
-        dataValue.setValueForced(
-            ValidationUtils.normalizeBoolean( dataValue.getValue(),
-                valueContext.getDataElement().getValueType() ) );
+        String value = ValidationUtils.normalizeBoolean( dataValue.getValue(),
+            valueContext.getDataElement().getValueType() );
```

Comment:
FYI: Note that we previously would update the value as a "side-effect" of this validation. Since the `dataValue` is not truely read-only this can no longer occur here. Therefore I duplicated this updating of the value when we create the "internal" `DataValue` from the `DataValueEntry`.